File: skink.py
from lexer import Lexer
# from parser_ import Parser

# def run_string(text):
#     run('<anonymous>', text)

def run(file):
    contents = open(file, 'r').read()
    
    _run(file.split('/')[-1], contents)
    
def _run(text):
    lexer = Lexer(text) 
    # tokens = lexer.generate_tokens()
    
    # # parser = Parser(file, tokens)

    # # print(f'tree: {parser.parse()}')

File: Skink.java
package com.github.skink;

import java.util.LinkedList;

public class Skink {
    private static void run(String file, String source) {
        Lexer lexer = new Lexer(file, new CharStream(source));
        LinkedList<Token> tokens = lexer.lex();

        Parser parser = new Parser(file, new TokenStream(tokens));
        Node node = parser.parse();

        System.out.println("tree:");
        System.out.println(node);
    }
    
    public static void runString(String source) {
        run("<anonymous>", source);
    }
}